app/Http/Request/ContactRequest.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace App\Http\Request;
+
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Validation\Rule;
+
+
+class ContactRequest extends FormRequest
+{
+    public function authorize(): bool
+    {
+        return true;
+    }
+
+    public function rules(): array
+    {
+        return[
+            'firstname' => ['min:2', 'required', 'string'],
```

Comment:
I recommend required to be first one, string second and after remaining ones.

Reply:
+